engine/src/test/java/io/zeebe/engine/processing/variable/mapping/MessageOutputMappingTest.java

```diff
@@ -183,32 +183,35 @@ public void shouldApplyOutputMappings() {
             .getFirst()
             .getKey();
 
-    final Record<VariableRecordValue> initialVariable =
+    final long initialVariablePosition =
         RecordingExporter.variableRecords(VariableIntent.CREATED)
             .withWorkflowInstanceKey(workflowInstanceKey)
-            .withName(CORRELATION_VARIABLE)
-            .getFirst();
+            .filter(r -> variables.containsKey(r.getValue().getName()))
+            .limit(variables.size())
+            .map(Record::getPosition)
+            .max(Comparator.naturalOrder())
+            .orElseThrow();
```

Comment:
How is this the `initialVariablePosition` if you take the max position? Isn't this the position of the last written `Variable:Created` event?

It's not clear to me why the test is changed to check records from a later position than before.

Reply:
Yeah the name is wrong, I should also change that. What happens is that the test was relying on a specific ordering of when events would be written to the log, e.g. the variable record for "i" first, then the variable record for the `CORRELATION_VARIABLE` after. So it would just grab the latter, and look for variable records that came after it as "new" variables.

Since we use the document indexing stuff everywhere now, which under the hood uses a map, the ordering isn't well defined, so you really want to grab both variables and use the one that came last as your lower bound.

Tbh I'm not a fan of how the test is written, but if we want to rewrite them, we should do that probably once the variable stuff has been migrated.